benchmarks/tosi_et_al_2015_gcubed/tosi.cc

```diff
@@ -0,0 +1,621 @@
+#include <aspect/geometry_model/interface.h>
+#include <aspect/global.h>
+#include <aspect/geometry_model/box.h>
+#include <aspect/postprocess/interface.h>
+//#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+
+namespace aspect
+{
+  /**
+   * This is the material model used for the viscoplastic convection
+   * benchmark models that were presented in the open access Tosi et al. (2015) paper:
+   * @code
+   *  @Article{T15,
+   *   Author = {Tosi, N. and Stein, C. and Noack, L. and H\"uttig, C. and Maierova, P. and Samual, H. and Davies, D. R. and Wilson, C. R. and Kramer, S. C. and Thieulot, C. and Glerum, A. and Fraters, M. and Spakman, W. and Rozel, A. and Tackley, P. J.},
+   *    Title = {A community benchmark for viscoplastic thermal convection in a 2-D square box},
+   *    Journal = {Geochemistry, Geophysics, Geosystems},
+   *    Year = {2015}
+   *    Pages = {2175-2196},
+   *    Volume = {16},
+   *    Doi = {10.1002/2015GC005807}}
+   * @endcode
+   *
+   * It is a material model that consists of globally constant values for all
+   * material parameters except density and viscosity. Density is temperature dependent,
+   * but practically constant through a small value of the thermal expansivity,
+   * and viscosity depends on temperature, depth and strain rate.
+   *
+   * The model is considered incompressible, following the definition
+   * described in Interface::is_compressible.
+   *
+   * @ingroup MaterialModels
+   */
+  namespace TosiBenchmark
+  {
+    using namespace dealii;
+
+
+    /**
+     * @note This benchmark only talks about the flow field, not about a
+     * temperature field. All quantities related to the temperature are
+     * therefore set to zero in the implementation of this class.
```

Comment:
I think this note would better fit the general comment above -- or at least put the two together in one place